(DOCSP-7078): Add support files page (#19)
mongodb/docs-realm: source/data-model/support-files.txt

or Realm Studio. In advanced cases, you may need to manually
delete the support files. Otherwise, these are
implementation details that you can safely ignore, presented
here only for the curious.

Support File Types
------------------

Realm's internal operation depends on several support file
types:

``<path>.realm``:
  The **realm file** contains the data and metadata of the
  realm at the given :ref:`Realm path <realm-path>`. Each
  app may have several realms at different paths, each with
  their own realm file.

``<name>.realm.lock``:
  The **lock file** enables synchronization between writes.

``<name>.realm.note``:
  The **note file** is a :wikipedia:`named pipe
  <Named_pipe>` for inter-thread and inter-process
  notifications. While usually completely internal, it has
  been known to cause the "unsupported resource found" error
  in iOS builds when it is accidentally added to the app
  bundle, e.g. when bundling a realm file with the app. If
  that happens, delete the pipe.

``<name>.realm.management``:
  The **management directory** contains internal files for
  the realm's state management.

Manually Deleting Support Files
-------------------------------

> In some circumstances, you may need to delete the support

Reviewer: ```suggestion
In some circumstances, you might need to delete the support
```

Reviewer: In which circumstances?

files. You may delete support files when the realm is
closed. Deleting any of a realm's files while the realm is

> open may corrupt the realm or disrupt :doc:`Sync

Reviewer: ```suggestion
open might corrupt the realm or disrupt :doc:`Sync
```

</sync/sync-overview>`.

Client Reset Scenario
~~~~~~~~~~~~~~~~~~~~~

When using :doc:`Realm Sync </sync/sync-overview>`, a **client
reset** is a serious error recovery task that your client
app must perform in the following situation:

> - The given synced realm on the server was restored from a backup, e.g. due to a Realm server crash.

Reviewer: ```suggestion
- The given synced realm on the server was restored from a backup. For example, due to a Realm server crash.
```
Avoid Latin phrases/abbreviations because they can be difficult for ESL readers and translation.

Reviewer: Wrap text to 72(?) characters

> - The client app made changes to that realm since the backup was made, but did not sync those changes back to the server before the server was restored.

Reviewer: Text wrap.

Author: I don't hard wrap on bullet points because the VSCode extension does not handle it properly.

In other words, the client app must carry out a client reset
on a given synced realm if the server is restored to a

> version before the version on the client.

Reviewer: ```suggestion
version earlier than the version on the client.
```
or 
```suggestion
version older than the version on the client.
```
"Before" has too many connotations around the order in which to do things, and is a little confusing here.

> On the client app side, the client reset procedure is as follows:

Reviewer: This is a little unclear to me. Is the app performing these steps, or is the user? This procedure falls under "Manually Deleting Support Files", which I assumed meant the user would be doing the deleting. Either way, the intro sentence and steps need to align with whoever is doing the action. For example:
```
During a client reset, the app:
1. Creates a backup of the local realm file.
2. Deletes the local realm file.
3. Downloads a fresh copy of the realm file the next time the app 
   connects to the server and opens that realm. The local changes
   made since the backup on the server are not reflected in the new
   file.
```
or
```
During a client reset, you must:
1. Create a backup of the local realm file.
2. Delete the local realm file.
The next time the app connects to the server and opens that realm, 
the app dowloads a fresh copy of the realm file. The local changes 
made since the backup on the server are not reflected in the new file.
```
If it is the app doing the deleting and not the user, I'd probably remove "Manually" from the section header.

Author: The user (programmer) is programming their app to do it.

1. Create a backup of the local realm file.
#. Delete the local realm file.

> #. The next time the app connects to the server and opens that realm, the app will download a fresh copy of the realm file. The local changes made since the backup on the server are not reflected in the new file.

Reviewer: ```suggestion
#. The next time the app connects to the server and opens that realm, the app downloads a fresh copy of the realm file. The local changes made since the backup on the server are not reflected in the new file.
```
Use present tense, as future tense can imply uncertainty. [Reference](https://docs-mongodbcom-staging.corp.mongodb.com/meta/jdestefano/style-guide-v3/style-guide/writing/use-present-tense.html#use-present-tense)

Author: I would have, but in this case the sentence starts with "The next time...", which made using present tense there sound grammatically wrong.
